bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/items/ItemUpdater.java

```diff
@@ -67,7 +79,14 @@ protected void receiveUpdate(ItemStateEvent updateEvent) {
                     }
                 }
                 if (isAccepted) {
+                    State oldState = item.getState();
                     item.setState(newState);
+                    if (oldState != null && !oldState.equals(newState)) {
```

Comment:
There were many complaints by openHAB users when this was once the case; most people use some "restoreOnStartup" policy, so they simply expect the system after a restart to continue to work and not to have an event for every single item on startup.

But nonetheless, the line looks a bit buggy to me. A state should usually never be `null`, but it is rather `UnDefType.NULL` at startup, see https://github.com/openhab/openhab2/blob/master/bundles/core/org.openhab.core.compat1x/src/main/java/org/openhab/core/items/GenericItem.java#L42.